File: angular/src/app/models/response/ville-response.ts
export class VilleResponse {
    public get ambiance(): string {
        return this._ambiance;
    }
    public set ambiance(value: string) {
        this._ambiance = value;
    }
    public get nom(): string {
        return this._nom;
    }
    public set nom(value: string) {
        this._nom = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    constructor(
        private _id: number,
        private _nom: string,
        private _ambiance: string
    ) { }
}

File: spring/src/main/java/fr/formation/api/response/VilleResponse.java
package fr.formation.api.response;

import org.springframework.beans.BeanUtils;

import fr.formation.model.Ville;

public class VilleResponse {

	private int id;
	private String nom;
	private String ambiance;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getAmbiance() {
		return ambiance;
	}

	public void setAmbiance(String ambiance) {
		this.ambiance = ambiance;
	}

	public static VilleResponse convert(Ville ville) {
		VilleResponse response = new VilleResponse();
		BeanUtils.copyProperties(ville, response);
		return response;
	}

}
